openmetadata-ui/src/main/resources/ui/src/pages/KPIPage/AddKPIPage.tsx

```diff
@@ -229,227 +228,224 @@ const AddKPIPage = () => {
+                rules={[
+                  {
+                    required: true,
+                    message: t('message.field-text-is-required', {
+                      fieldText: t('label.data-insight-chart'),
+                    }),
+                  },
+                ]}>
+                <Select
+                  data-testid="dataInsightChart"
+                  notFoundContent={t('message.all-charts-are-mapped')}
+                  placeholder={t('label.select-a-chart')}
+                  value={selectedChart?.fullyQualifiedName}
+                  onChange={handleChartSelect}>
+                  {chartOptions.map((chart) => (
+                    <Option key={chart.fullyQualifiedName}>
+                      {chart.displayName || chart.name}
+                    </Option>
+                  ))}
+                </Select>
+              </Form.Item>
 
-                <Form.Item label={t('label.display-name')} name="displayName">
-                  <Input
-                    data-testid="displayName"
-                    placeholder={t('label.kpi-display-name')}
-                    type="text"
-                  />
-                </Form.Item>
+              <Form.Item label={t('label.display-name')} name="displayName">
+                <Input
+                  data-testid="displayName"
+                  placeholder={t('label.kpi-display-name')}
+                  type="text"
+                />
+              </Form.Item>
 
+              <Form.Item
+                label={t('label.select-a-metric-type')}
+                name="metricType"
+                rules={[
+                  {
+                    required: true,
+                    message: t('message.field-text-is-required', {
+                      fieldText: t('label.metric-type'),
+                    }),
+                  },
+                ]}>
+                <Select
+                  data-testid="metricType"
+                  disabled={isUndefined(selectedChart)}
+                  placeholder={t('label.select-a-metric-type')}
+                  value={selectedMetric?.name}
+                  onChange={handleMetricSelect}>
+                  {metricTypes.map((metric) => (
+                    <Option key={metric.name}>
+                      {`${metric.name} (${metric.chartDataType})`}
+                    </Option>
+                  ))}
+                </Select>
+              </Form.Item>
+
+              {!isUndefined(selectedMetric) && (
                 <Form.Item
-                  label={t('label.select-a-metric-type')}
-                  name="metricType"
+                  label={t('label.metric-value')}
+                  name="metricValue"
                   rules={[
                     {
                       required: true,
-                      message: t('message.field-text-is-required', {
-                        fieldText: t('label.metric-type'),
-                      }),
+                      validator: () => {
```

Comment:
we can use min max for this validation i believe

Reply:
can we take care of this later ?